airflow/jobs/job.py

```diff
@@ -252,16 +252,16 @@ def most_recent_job(job_type: str, session: Session = NEW_SESSION) -> Job | None
     :param job_type: job type to query for to get the most recent job for
     :param session: Database session
     """
-    return (
-        session.query(Job)
-        .filter(Job.job_type == job_type)
+    return session.scalars(
+        select(Job)
+        .where(Job.job_type == job_type)
         .order_by(
             # Put "running" jobs at the front.
             case({State.RUNNING: 0}, value=Job.state, else_=1),
             Job.latest_heartbeat.desc(),
         )
-        .first()
-    )
+        .limit(1)
+    ).first()
```

Comment:
```python
return session.scalar(
    select(Job)
    .where(Job.job_type == job_type)
    .order_by(...)
    .limit(1)
)
```

I think this would be equivalent?